pkgs/development/python-modules/pytest-bdd/default.nix

```
@@ -0,0 +1,23 @@
+{ stdenv, buildPythonPackage, fetchPypi, glob2, Mako, parse, parse-type, py, pytest, six }:
+
+buildPythonPackage rec {
+  pname = "pytest-bdd";
+  version = "3.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1v5dazwmsz2rvhbwg1i4y9qnn0ch1yzflca9c7n93sq2f2603xiy";
+  };
+
+  propagatedBuildInputs = [ glob2 Mako parse parse-type py pytest six ];
+
+  # Tests require extra dependencies
+  doCheck = false;
```

that fine... just add them to checkInputs
```suggestion
  checkInputs = [ mock ];
```

---

After running the following command:

```
find pkgs/development/python-modules -name default.nix | xargs grep doCheck | grep pytest
```

I suspect we really need that *doCheck = false*.

---

no... `doCheck = false;` will make it skip the checkPhase entirely

Here's how i would do this expression:
```
{ stdenv, buildPythonPackage, fetchFromGitHub
, execnet
, glob2
, Mako
, mock
, parse
, parse-type
, py
, pytest
, six
}:

buildPythonPackage rec {
  pname = "pytest-bdd";
  version = "3.2.1";

  # tests are not included in pypi tarball
  src = fetchFromGitHub {
    owner = "pytest-dev";
    repo = pname;
    rev = version;
    sha256 = "02y28l5h1m9grj54p681qvv7nrhd7ly9jkqdchyw4p0lnmcmnsrd";
  };

  propagatedBuildInputs = [ glob2 Mako parse parse-type py pytest six ];

  checkInputs = [ execnet mock pytest ];
  checkPhase = ''
    pytest
  '';

  meta = with stdenv.lib; {
    description = "BDD library for the py.test runner";
    homepage = https://github.com/pytest-dev/pytest-bdd;
    license = licenses.mit;
    maintainers = with maintainers; [ jm2dev ];
  };
}
```

---

Thanks to that "fetchFromGitHub" now I'm aware of the available fetchers, and I suspect you  have plenty of experience with python pips.

---

I'm a co-owner of the nixpkgs python packages, only done a few hundred